mixed-content/generic/mixed-content-test-case.js

```diff
@@ -0,0 +1,224 @@
+function MixedContentTestCase(scenario, description, sanityChecker) {
+  var insecureProtocol = "http:"
+  var secureProtocol = "https:"
+
+  var sameOriginHost = location.hostname;
+  var crossOriginHost = "{{domains[www1]}}";
+
+  var insecurePort = getNormalizedPort(parseInt("{{ports[http][0]}}"));
+  var securePort = getNormalizedPort(parseInt("{{ports[https][0]}}"));
+
+  var resourcePath = "/mixed-content/generic/expect.py";
+
+  var endpoint = {
+    "same-origin":
+      location.origin + resourcePath,
+    "same-host-https":
+      secureProtocol + "//" + sameOriginHost + securePort + resourcePath,
```

Comment:
Yes, `getNormalizedPort` prepends a colon. If a default or no port is used, it evaluates to an empty string. Added a JsDoc to the func. 

Scope of securePort and insecurePort is fairly small, however I am not sure how to name it properly: E.g. secureMaybePort, securePortish, securePortClause, securePortPortion all sound pretty bad. I suppose I'll just add a comment what the values can be.